benchmarking.ipynb

```diff
@@ -0,0 +1,428 @@
+{
```

Comment:
<p>timeit is nice - but we should store the output to a file, that we can save it</p><p> </p><p>We need a warmup phase as well (first requests are normally slow). We need something like</p><p> </p><pre class="ql-syntax" spellcheck="false">import time


MODEL_NAME_PT = "t4r_pytorch_pt"


# WarmUp
for _ in range(200):
&nbsp; &nbsp; payload = cudf.DataFrame(data={'sess_pid_seq': np.random.randint(0, 390001, 20), 'id': 0}).groupby('id').agg({'sess_pid_seq': list})
&nbsp; &nbsp; with grpcclient.InferenceServerClient("localhost:8001") as client:
&nbsp; &nbsp; &nbsp; &nbsp; col_names = ['sess_pid_seq']
&nbsp; &nbsp; &nbsp; &nbsp; inputs = nvt_triton.convert_df_to_triton_input(col_names, payload, grpcclient.InferInput)
&nbsp; &nbsp; &nbsp; &nbsp; response = client.infer(MODEL_NAME_PT, inputs)
&nbsp; &nbsp; end_time = time.time()


# Collecting
out = []
for _ in range(200):
&nbsp; &nbsp; payload = cudf.DataFrame(data={'sess_pid_seq': np.random.randint(0, 390001, 20), 'id': 0}).groupby('id').agg({'sess_pid_seq': list})


&nbsp; &nbsp; start_time = time.time()
&nbsp; &nbsp; with grpcclient.InferenceServerClient("localhost:8001") as client:
&nbsp; &nbsp; &nbsp; &nbsp; col_names = ['sess_pid_seq']
&nbsp; &nbsp; &nbsp; &nbsp; inputs = nvt_triton.convert_df_to_triton_input(col_names, payload, grpcclient.InferInput)
&nbsp; &nbsp; &nbsp; &nbsp; response = client.infer(MODEL_NAME_PT, inputs)
&nbsp; &nbsp; end_time = time.time()
&nbsp; &nbsp; out.append(end_time-start_time)
</pre><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Transformers4Rec/pull/615/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>